lib/BridgedRoom.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+function BridgedRoom(bridge, gitter, gitterRealtime, matrixRoomModel, gitterRoomModel) {
```

Comment:
Style: When you get beyond about 3 args, it becomes a cognitive burden to remember which arg goes where. JS doesn't help you there as it does not provide python-style kwargs like:
```python
output = foo(bar=5, baz="quuz")
```

However, you can emulate this by the "options" object (typically called `opts` for succinctness). This is a plain JS object which has the argument names as keys:
```javascript
function BridgedRoom(opts) {
  this._bridge = opts.bridge;
  this._gitter = opts.gitter;
  // ...
}

room = new BridgedRoom({ bridge: b, gitter: g });
```

Apart from being much easier to grok when calling the method, you now also have a way to extend the function **without breaking backward-compat**, because you can just add more keys. I advise you use `opts` liberally, particularly when you have a function which has so many args like this one.